presto-main/src/test/java/com/facebook/presto/block/AbstractTestBlock.java

```diff
@@ -86,6 +90,31 @@
         }
     }
 
+    protected List<Block> splitBlock(Block block, int count)
+    {
+        double sizePerSplit = block.getPositionCount() * 1.0 / count;
+        ImmutableList.Builder<Block> result = ImmutableList.builder();
+        for (int i = 0; i < count; i++) {
+            int startPosition = toIntExact(Math.round(sizePerSplit * i));
+            int endPosition = toIntExact(Math.round(sizePerSplit * (i + 1)));
+            result.add(block.getRegion(startPosition, endPosition - startPosition));
+        }
+        return result.build();
+    }
+
+    private void assertBlockSize(Block block)
+    {
+        // Asserting on `block` is not very effective because most blocks passed to this method is compact.
+        // Therefore, we split the `block` into two and assert again.
+        assertEquals(block.getSizeInBytes(), copyBlock(block).getSizeInBytes());
+
+        List<Block> splitBlock = splitBlock(block, 2);
+        Block firstHalf = splitBlock.get(0);
+        Block secondHalf = splitBlock.get(1);
+        assertEquals(firstHalf.getSizeInBytes(), copyBlock(firstHalf).getSizeInBytes());
+        assertEquals(secondHalf.getSizeInBytes(), copyBlock(secondHalf).getSizeInBytes());
```

Comment:
We discussed offline and decided we are not adding this:
* It's not high signal enough
* It would fail for blocks like RLE block